### PR TITLE
feat: Handle Safari-mangled backup ZIPs and improve backup UI

### PR DESCRIPTION
**Pull request by michael-genson**
## What type of PR is this?

_(REQUIRED)_

- feature

## What this PR does / why we need it:

_(REQUIRED)_

We've had several instances [like this](https://github.com/mealie-recipes/mealie/issues/1500#issuecomment-2142960420) where Safari mangles the ZIP file structure, breaking our backups. This PR adds some logic to detect when this happens and work around it.

The UI is also very unhelpful when backups fail, so this PR fixes some state issues and flashes an alert to check server logs when a backup fails.

## Which issue(s) this PR fixes:

_(REQUIRED)_

No open issues, several closed ones where the solution was just to fix the ZIP structure

## Testing

_(fill-in or delete this section)_

I added a mangled zip file test before implementing the workaround, confirmed it failed, and now it passes

### CODE REVIEW

File: tests/data/__init__.py [hunk 1]
```diff
@@ -22,6 +22,9 @@
 backup_version_bcfdad6b7355_1 = CWD / "backups/backup_version_bcfdad6b7355_1.zip"
 """bcfdad6b7355: remove tool name and slug unique contraints"""
 
+backup_version_09aba125b57a_1 = CWD / "backups/backup_version_09aba125b57a_1.zip"
+"""09aba125b57a: add OIDC auth method (Safari-mangled ZIP structure)"""
```
Comment (boc-the-git) [inline tests/data/__init__.py:26]: This might be a silly question, but what's the relevance of `add OIDC auth method`?? 😕
Appears in a couple files.

**Review by boc-the-git — APPROVED** [review]
This LGTM, though I am a bit curious re my other comment.
Assuming there's no cause for concern there, happy for you to merge @michael-genson 👍

File: tests/data/__init__.py (hunk 1 above)
Comment (michael-genson) [reply tests/data/__init__.py:26]: The original purpose of having a bunch of example backup files in the tests data was to test for migration issues, so for all of the migration test data I include the alembic version (`09aba125b57a` in this case) as well as the migration message. This is basically saying "this backup is a snapshot of a database that is at the migration where the OIDC auth method was added".

For this specific test it's kind of useless, though in the future if this migration fails it's easier to debug

Comment (boc-the-git) [reply tests/data/__init__.py:26]: Ah, makes sense. I tried to check if it was a commit hash and obviously it wasn't! Didn't think of alembic

**Pull request merged**